implementation/test/DepositFraudTest.js

```diff
@@ -341,6 +344,102 @@ describe("DepositFraud", async function() {
     })
   })
 
+  describe("startSignerFraudLiquidation", async () => {
+    let signerBond
+    before(async () => {
+      signerBond = 10000000
+      await ecdsaKeepStub.send(signerBond, {from: accounts[1]})
+    })
+
+    beforeEach(async () => {
+      await createSnapshot()
+    })
+
+    afterEach(async () => {
+      await restoreSnapshot()
+    })
+
+    it("executes and emits StartedLiquidation event", async () => {
+      const block = await web3.eth.getBlock("latest")
+
+      await testDeposit.startSignerFraudLiquidation({from: owner})
+
+      const events = await tbtcSystemStub.getPastEvents("StartedLiquidation", {
+        fromBlock: block.number,
+        toBlock: "latest",
+      })
+
+      const initiator = await testDeposit.getLiquidationInitiator()
+      const initiated = await testDeposit.getLiquidationTimestamp()
+
+      expect(events[0].returnValues[0]).to.equal(testDeposit.address)
+      expect(events[0].returnValues[1]).to.be.true
+      expect(events[0].returnValues[2]).to.eq.BN(block.timestamp)
+
+      expect(initiator).to.equal(owner)
+      expect(initiated).to.eq.BN(block.timestamp)
+    })
+
+    it("completes liquidation if we came from the redemption flow", async () => {
```

Comment:
Is this really, “it liquidates immediately with bonds going to the redeemer if we came from the redemption flow”?

Reply:
Also… Is this correct behavior? Shouldn't we require fraud proof before liquidating for signer fraud?

Reply:
> Also… Is this correct behavior? Shouldn't we require fraud proof before liquidating for signer fraud?

We do in contracts. `startSignerFraudLiquidation` is only called at the end of ECDSA/SPV fraud proof.

Reply:
I think what confused me here is that the guard to make sure this is allowed to be called happens at a higher level. Makes sense with the way that these contracts fit together though.